src/lib.rs

```diff
@@ -91,13 +95,13 @@ impl Agent {
         }
     }
 
-    async fn send_metric(&self, val: MetricValue) {
-        let metric = HostMetricWrapper(&self.host_id, val).into();
-        // TODO: error handling.
-        let result = self.client.post_metrics(metric).await;
-        if result.is_err() {
-            dbg!(result.err());
-        }
+    async fn send_metric(&mut self, val: MetricValue) {
+        let mut metric: Vec<_> = HostMetricWrapper(&self.host_id, val).into();
+        metric.extend(self.metric.clone());
+        let result = self.client.post_metrics(metric.clone()).await;
+        // If Ok, then heaped metric must be empty, else extend it.
+        // TODO: Drop metric if it is too ancient.
+        self.metric = if result.is_ok() { vec![] } else { metric };
```

Comment:
Instead of silently fails, It's better to log a warning here.

Reply:
Yes, we need logger....:innocent: